pinot-core/src/main/java/org/apache/pinot/core/util/TableConfigUtils.java

```diff
@@ -76,10 +79,41 @@ public static void validate(TableConfig tableConfig, @Nullable Schema schema) {
     validateFieldConfigList(tableConfig.getFieldConfigList(), schema);
   }
 
+  /**
+   * Validates all table config including tenant config. This is specifically invoked
+   * during table config creation, update or validate API calls.
+   */
+  public static void validate(TableConfig tableConfig, @Nullable Schema schema, Set<String> serverTenants,
+      Set<String> brokerTenants) {
+    validate(tableConfig, schema);
+    validateTenants(tableConfig, serverTenants, brokerTenants);
+  }
+
+  /**
+   * Ensures a valid tenant config is provided in the table config.
+   * Ensures the broker and server tenant names are valid.
+   */
+  private static void validateTenants(TableConfig tableConfig, Set<String> serverTenants, Set<String> brokerTenants) {
```

Comment:
This is the same check as `PinotHelixResourceManager.validateTableTenantConfig(TableConfig tableConfig)`